feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,745 @@
+		} else {
+			if root.GetCounters().GetInFcsErrors() > 0 {
+				t.Errorf("ERROR: Counter InFcsErrors is %d, not zero", root.GetCounters().GetInFcsErrors())
+			}
+		}
+	}
+}
+
+func TestInterfacesubIntfs(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	for _, intf := range interfaces {
+		subIntfIndexes := gnmi.LookupAll(t, dut, gnmi.OC().Interface(intf).SubinterfaceAny().Index().State())
+		for _, index := range subIntfIndexes {
+			subIntfIndex, present := index.Val()
+			if !present {
+				t.Errorf("ERROR: subIntf index value doesn't exist")
+				continue
+			}
+
+			subIntfPath := gnmi.OC().Interface(intf).Subinterface(subIntfIndex)
+			subIntfState := gnmi.Get(t, dut, subIntfPath.State())
+			subIntf := subIntfState.GetName()
+			intfState := gnmi.Get(t, dut, gnmi.OC().Interface(intf).State())
+			if subIntfState.OperStatus != oc.Interface_OperStatus_UP {
+				t.Errorf("ERROR: Oper status is not up on interface %s", subIntf)
+			}
+
+			if subIntfState.AdminStatus != oc.Interface_AdminStatus_UP {
+				t.Errorf("ERROR: Admin status is not up on interface %s", subIntf)
+			}
+
+			if subIntfState.Description == nil {
+				t.Errorf("ERROR: Description is not present on interface %s", subIntf)
+			}
+
+			if subIntfState.GetCounters().OutOctets == nil && intfState.GetCounters().OutOctets == nil {
+				t.Errorf("ERROR: Counter OutOctets is not present on interface %s, %s", subIntf, intf)
+			}
+
+			if subIntfState.GetCounters().InMulticastPkts == nil && intfState.GetCounters().InMulticastPkts == nil {
+				t.Errorf("ERROR: Counter InMulticastPkts is not present on interface %s", subIntf)
+			}
+
+			counters := subIntfPath.Counters()
+			parentCounters := gnmi.OC().Interface(intf).Counters()
+
+			cases := []struct {
+				desc          string
+				counter       ygnmi.SingletonQuery[uint64]
+				parentCounter ygnmi.SingletonQuery[uint64]
+			}{
+				{
+					desc:          "InDiscards",
+					counter:       counters.InDiscards().State(),
+					parentCounter: parentCounters.InDiscards().State(),
+				},
+				{
+					desc:          "InErrors",
+					counter:       counters.InErrors().State(),
+					parentCounter: parentCounters.InErrors().State(),
+				},
+				{
+					desc:          "InUnknownProtos",
+					counter:       counters.InUnknownProtos().State(),
+					parentCounter: parentCounters.InUnknownProtos().State(),
+				},
+				{
+					desc:          "OutDiscards",
+					counter:       counters.OutDiscards().State(),
+					parentCounter: parentCounters.OutDiscards().State(),
+				},
+				{
+					desc:          "OutErrors",
+					counter:       counters.OutErrors().State(),
+					parentCounter: parentCounters.OutErrors().State(),
+				},
+				{
+					desc:          "InFcsErrors",
+					counter:       counters.InFcsErrors().State(),
+					parentCounter: parentCounters.InFcsErrors().State(),
+				},
+			}
+
+			for _, c := range cases {
+				t.Run(c.desc, func(t *testing.T) {
+
```

Comment:
Delete blank line

Reply:
Thanks for your comments. Removed extra blank line.
```go
t.Run(c.desc, func(t *testing.T) {
					if val, present := gnmi.Lookup(t, dut, c.counter).Val(); present && (val == 0) {
```